config/mongoid.yml

```diff
@@ -8,6 +8,7 @@ common: &default_client
     retry_interval: <%= ENV['MONGOID_RETRY_INTERVAL'] || 0 %>
     timeout: <%= ENV['MONGOID_TIMEOUT'] || 0.5 %>
     ssl: <%= ENV['MONGOID_USE_SSL'] || false %>
+    auth_source: <%= ENV['MONGOID_AUTH_SOURCE'] || '' %>
```

Comment:
@xitij2000, this looks okay, but will passing `''` as the default value work okay? From what I can see `admin` is the default value used by `mongoid`.

Reply:
@lgp171188 I am no expert on Mongo, but what I've learned so far about this is that users can be created in any database. The approach edx-platform and the forum expect is that the users should be created in the same database used by the application, i.e. the edxapp database for edx-platform. If you don't specify an auth_source that seems to be the approach it uses (I have tested this). Perhaps this is a change between mongoid 5.0 and 5.2, but leaving this blank allows working with our existing mongo setup, but leaving it blank doesn't work with Atlas.

Reply:
@xitij2000, not sure if I understand your response - my question was about using `''` as the fallback when the default used by mongoid is `admin`. Can you clarify if misunderstood something?

Reply:
@lgp171188 The default for *mongoid* is admin, however the default for *cs_comments_service* isn't. By defaulting it to admin I would be changing the default behaviour for the forum which would break existing use cases. So it's set to an empty string which makes mongoid use the current database for autentication.

Reply:
@xitij2000, thanks for checking and verifying. :+1: